x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -2142,6 +2212,130 @@ private static Expression typeSpecificConvert(ConvertFunction convert, Source so
         }
     }
 
+    private static class ResolveUnmapped extends ParameterizedAnalyzerRule<LogicalPlan, AnalyzerContext> {
```

Comment:
thought: as the analyzer grows, we should probably start putting analyzer rules into their own files. It's fine for this PR, of course.

Reply:
Yes, I had the same thought even before this PR; we do the same with the optimizer rules, don't think we need this huge class. I've applied the suggestion. I will follow up with breaking the class down (at some point).